test/baseline/file/FileUpload.test.js

```diff
@@ -0,0 +1,413 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-env mocha */
+
+// ****************************************************************
+//  THIS FILE IS USED TO GENERATE TEST DATA FOR THE files-seeder.js
+//    SEQUELIZE SEEDER.
+//  IT IS NOT A LEGITIMATE TEST FILE
+// ****************************************************************
+//  DO NOT RUN THIS FILE UNLESS YOU WANT TO GENERATE NEW TEST DATA
+// ****************************************************************
+
+const { request, expect, testArtistUserId, testAccessToken } = require('../../testConfig')
+const MockAccessToken = require('../../MockAccessToken')
+
+const audioSourceFilePath = '/var/www/api/test/media/audio/'
+
+let response = null
+let trackId = null
+let fileName = null
+
+describe.skip('baseline / file upload endpoint tests', () => {
+  MockAccessToken(testArtistUserId)
+
+  it('should reject a PUT request without access token', async () => {
+    response = await request.put('/user/tracks/:id/file')
+
+    expect(response.status).to.eql(401)
+  })
+  it('should reject a PUT request with with a valid access token, but no track id ', async () => {
+    response = await request.put('/user/tracks/:id/file')
+      .set('Authorization', `Bearer ${testAccessToken}`)
+
+    expect(response.status).to.eql(500)
+  })
+  it('should reject a PUT request with with a valid access token, but invalid track id ', async () => {
+    response = await request.put('/user/tracks/1234-asdf/file')
+      .set('Authorization', `Bearer ${testAccessToken}`)
+
+    expect(response.status).to.eql(500)
+  })
+})
+
+describe.skip('ALBUM: Best Album Ever', () => {
+  MockAccessToken(testArtistUserId)
+
+  // Ergonomic interactive concept,  Laurie Yost
+  it('should process audio file / track 01', async () => {
+    trackId = '44a28752-1101-4e0d-8c40-2c36dc82d035'
+    fileName = 'whiteNoise5s1.m4a'
+
+    response = await request.put(`/user/tracks/${trackId}/file`)
+      .set('Authorization', `Bearer ${testAccessToken}`)
+      .attach('files', `${audioSourceFilePath}${fileName}`)
+
+    expect(response.status).to.eql(200)
+  })
+
+  // Customer-focused fresh-thinking groupware Miss Lewis Ondricka
+  it('should process audio file / track 02', async () => {
+    trackId = '706cff12-ba44-49f7-8982-98b3996a2919'
+    fileName = 'whiteNoise5s2.m4a'
+
+    response = await request.put(`/user/tracks/${trackId}/file`)
+      .set('Authorization', `Bearer ${testAccessToken}`)
+      .attach('files', `${audioSourceFilePath}${fileName}`)
+
+    expect(response.status).to.eql(200)
```

Comment:
How is it checking that the file actually exists? 

Is this cleaning up uploads afterwards? If not, will this just create new files every single time tests are run?

Reply:
Oh I just read the comment at the top. This should probably not be named a `test.js` file right if it is a seed script? Otherwise the tests will automatically run it.

Reply:
FileUpload.test.js is not a legitimate test file. It's only used to generate all of the test data. It shouldn't be run as a test file, as mentioned in the code comments and the 'HowTheFileUploadDataWasMade' file.

There is no cleanup, since we need the audio files to provide sound when the user clicks on a track.

My understanding of the issue was that we needed to seed the test/dev Docker container with everything needed to allow a user to click on a track and hear the associated audio. Basically this issue was about just adding more test data, similar to how we already seed the database.

Reply:
Gotcha, that makes sense. 

In that case though, does it make sense for it to be in the test directory? It's not a test. Couldn't we just run it as a js script? Does it need to run inside of mocha to use supertest for it, is that why it needs to be in the test directory? My worry is that right now it might accidentally get run as part of the test suite, just because of its location and format. Then people would get into a situation where they're indefinitely running the tests. 

I'm seeing 30 tests, which makes sense for the amount of tracks on the albums. Could you point to me where there's another 30 or so files needed?

Reply:
Ii am not sure about what to do with the test file at this point. I'd actually like to get rid of it, now that we have the needed data. This is for the reason/s you already mention.

However, it would be needed in the future if, for some reason, the test data had to be re-created. Also, I prefer to keep everything that contributes to the codebase. It's a tool, albeit one that wouldn't / shouldn't be run often, if at all.

It doesn't have to be in the test directory. Alternatively, it could remain where it is, and be excluded by adding a few lines to (I think) the mocha config file. For now, everything in the file is set to 'skip', so nothing in the file actually executes.

I don't understand what you mean by "another 30 or so files". We already have the needed files; I don't understand why we would need another 30.

Reply:
Ah, maybe "another 30 or so files" refers to the [trim] files.

Those are output by the upload process. I made as few assumptions as possible about what to do with all of the process' output, so I left the trim files in. They can probably go away.

Reply:
Ah so it's just a file that needs to get run once and you've already run it? I was thinking it had to be run as part of the seeding process. Honestly in that case I don't think it's needed. It'll just confuse people finding it in the future trying to figure out what it does.

Reply:
Yes, it's a 'one and done' kind of thing. Also, yes I've already run it.

Reply:
Ok, here's what I see as needing to change for this PR:

- Remove FileUpload.test.js and its folder
- Remove the 30-something [trim] files from test/media/audio
- Remove HowTheFileUploadDataWasMade.md folder, and any references to it

Let me know if this is correct, or if there's more to be done. Once you confirm, I can push the changes.

Reply:
Also, there's a broken test somewhere in the test suite, as the result of this PR. Although it's related, it looks like a separate matter. Once we sort out this PR and merge, I will fix the test/s,

Reply:
The test talks about using FileUpload.test.js. Since we decided to delete this file, it made little sense to me to keep anything talking about the test data creation.

Reply:
Re-added the trim files. There is now a separate PR for this issue.